astropy/time/core.py

```diff
@@ -583,6 +588,40 @@ def value(self):
         # the ``value`` attribute is cached.
         return getattr(self, self.format)
 
+    @property
+    def masked(self):
+        return self._time.masked
+
+    @property
+    def mask(self):
+        return self._time.mask
+
+    def __setitem__(self, item, value):
+        if value in (np.ma.masked, np.nan):
+            self._time.jd2[item] = np.nan
+            del self.cache
+            return
+
+        if not isinstance(value, Time):
+            try:
+                value = self.__class__(value, scale=self.scale, location=self.location)
+            except Exception:
+                try:
+                    value = self.__class__(value, scale=self.scale, format=self.format,
+                                           location=self.location)
+                except Exception:
+                    raise ValueError('cannot convert value to a compatible Time object')
+
+        if self.location != value.location:
```

Comment:
I think that a non-scalar location is a corner case and that we could just raise an exception if either the `self` or `value` location is not a scalar or `None`.  At least for now this would be fine and we can see if any users actually complain.